tests/core/full_node/stores/test_full_node_store.py

```diff
@@ -29,18 +29,21 @@
 from tests.util.keyring import TempKeyring
 
 
+test_constants = test_constants_original.replace(**{"DISCRIMINANT_SIZE_BITS": 32, "SUB_SLOT_ITERS_STARTING": 2 ** 12})
+log = logging.getLogger(__name__)
+
+
 def cleanup_keyring(keyring: TempKeyring):
     keyring.cleanup()
 
 
-temp_keyring = TempKeyring()
-keychain = temp_keyring.get_keychain()
-atexit.register(cleanup_keyring, temp_keyring)  # Attempt to cleanup the temp keychain
-test_constants = test_constants_original.replace(**{"DISCRIMINANT_SIZE_BITS": 32, "SUB_SLOT_ITERS_STARTING": 2 ** 12})
-bt = create_block_tools(constants=test_constants, keychain=keychain)
-
+@pytest_asyncio.fixture(scope="function")
+async def custom_block_tools() -> BlockTools:
+    temp_keyring = TempKeyring()
+    keychain = temp_keyring.get_keychain()
+    atexit.register(cleanup_keyring, temp_keyring)  # Attempt to cleanup the temp keychain
```

Comment:
Thanks for cleaning this up. I'm pretty sure that we can use the more recent pattern of:

```suggestion
    with TempKeyring() as keychain:
        yield await create_block_tools_async(constants=test_constants, keychain=keychain)
```

There's some cleanup tasks that the TempKeychain performs during __exit__.